mllib/src/main/scala/org/apache/spark/ml/classification/LogisticRegression.scala

```diff
@@ -451,11 +471,14 @@ private class LogisticAggregator(
    * @param label The label for this data point.
    * @param data The features for one data point in dense/sparse vector format to be added
    *             into this aggregator.
+   * @param sampleWeight The weight for over-/undersamples each of training sample.
    * @return This LogisticAggregator object.
    */
-  def add(label: Double, data: Vector): this.type = {
+  def add(label: Double, sampleWeight: Double = 1.0, data: Vector): this.type = {
     require(dim == data.size, s"Dimensions mismatch when adding new sample." +
       s" Expecting $dim but got ${data.size}.")
+    require(sampleWeight >= 0.0, s"sampleWeight, ${sampleWeight} has to be >= 0.0")
```

Comment:
This check should probably happen in MultiClassSummarizer since it already checks for label validity.